OnlineShop/WebApplication3/WebApplication3/Controllers/AccountController.cs

```diff
@@ -0,0 +1,125 @@
+                FirstName = model.FirstName,  // Ім'я користувача
+                LastName = model.LastName   // Прізвище користувача
+            };
+
+            // Спробуємо створити користувача з паролем
+            var result = await _userManager.CreateAsync(user, model.Password);
+
+            // Якщо користувача створено успішно
+            if (result.Succeeded)
+            {
+                // Виконуємо вхід користувача після реєстрації (якщо успішно зареєстровано)
+                await _signInManager.SignInAsync(user, isPersistent: false); // isPersistent: false означає, що сесія не буде зберігатися після закриття браузера
+                return RedirectToAction("Index", "Home"); // Перехід на головну сторінку після успішної реєстрації
+            }
+
+            // Якщо виникли помилки під час реєстрації, додаємо їх в модель (для відображення на формі)
+            foreach (var error in result.Errors)
+            {
+                ModelState.AddModelError(string.Empty, error.Description); // Додаємо помилки в модель для відображення користувачу
+            }
+        }
+
+        // Якщо модель не валідна або виникли помилки, повертаємо користувача назад на форму реєстрації
+        return View(model);
+    }
+
+    // Сторінка логіну (GET)
+    [HttpGet]
+    public IActionResult Login()
+    {
+        return View(); // Повертає форму логіну
+    }
+
+    // Обробка логіну користувача (POST)
+    [HttpPost]
+    [ValidateAntiForgeryToken] // Захист від CSRF атак
+    public async Task<IActionResult> Login(LoginViewModel model, string returnUrl = null)
+    {
+        returnUrl ??= Url.Content("~/"); // Якщо не передано параметр returnUrl, використовуємо головну сторінку
+
+        // Якщо модель валідна (всі поля введені правильно)
+        if (ModelState.IsValid)
+        {
+            // Пробуємо увійти за допомогою логіну та пароля
+            var result = await _signInManager.PasswordSignInAsync(model.Email, model.Password, model.RememberMe, false);
+
+            // Якщо вхід успішний
+            if (result.Succeeded)
+            {
+                return RedirectToLocal(returnUrl); // Перехід на потрібну сторінку після успішного логіну
+            }
+            else
+            {
+                // Якщо вхід не вдався, додаємо помилку в модель для відображення на формі
+                ModelState.AddModelError(string.Empty, "Невірний логін або пароль.");
+            }
+        }
+
+        // Якщо модель не валідна або вхід не вдалося здійснити, повертаємо користувача на сторінку логіну
+        return View(model);
+    }
+
+    // Вихід з системи (POST)
+    [HttpPost]
+    [ValidateAntiForgeryToken] // Захист від CSRF атак
+    public async Task<IActionResult> Logout()
+    {
+        await _signInManager.SignOutAsync(); // Виконуємо вихід з системи
+        return RedirectToAction("Index", "Home"); // Переходимо на головну сторінку після виходу
```

Comment:
we don't have index page, remove please.
We don't need logout func also, remove